src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -2,41 +2,70 @@
 
 import static java.util.Objects.requireNonNull;
 
+import java.util.List;
+
 import seedu.address.commons.core.Messages;
 import seedu.address.model.Model;
-import seedu.address.model.person.NameContainsKeywordsPredicate;
+import seedu.address.model.ReadOnlyNavigation;
+import seedu.address.model.lecture.LectureName;
+import seedu.address.model.module.CodeContainsKeywordsPredicate;
+import seedu.address.model.module.LectureNameContainsKeywordsPredicate;
+import seedu.address.model.module.ModuleCode;
+import seedu.address.model.module.VideoNameContainsKeywordsPredicate;
+import seedu.address.model.navigation.NavigationContext;
 
 /**
- * Finds and lists all persons in address book whose name contains any of the argument keywords.
+ * Finds and lists all modules/lectures/videos in Le Tracker current context
+ * whose moduleCode/lectureName/videoName contains any of the argument keywords.
  * Keyword matching is case insensitive.
  */
 public class FindCommand extends Command {
 
     public static final String COMMAND_WORD = "find";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose names contain any of "
-            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
-            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
-            + "Example: " + COMMAND_WORD + " alice bob charlie";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+        + ": Finds all modules/lectures/videos whose moduleCode/lectureName/videoName contain any of "
+        + "the specified keywords (case-insensitive) separated with a space "
+        + "and displays them as a list with index numbers.\n"
+        + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+        + "Example: \n"
+        + "1." + COMMAND_WORD + " CS2040S\n"
+        + "2." + COMMAND_WORD + " Week1 Week2\n"
+        + "3." + COMMAND_WORD + " Video1 Video2 Video3\n";
 
-    private final NameContainsKeywordsPredicate predicate;
+    private List<String> keywords;
 
-    public FindCommand(NameContainsKeywordsPredicate predicate) {
-        this.predicate = predicate;
+    public FindCommand(List<String> keywords) {
+        this.keywords = keywords;
     }
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.updateFilteredPersonList(predicate);
+        ReadOnlyNavigation navigation = model.getNavigation();
+        NavigationContext navigationContext = navigation.getCurrentContext();
```

Comment:
Should NavigationContext be used directly in this case? NavigationContext was designed only for use with nav commands and navigation system. 

Commands such as AddCommand allow user to explicitly specify context e.g `add L1 /mod CS2040S` alongside context-based usage e.g. `add L1` from CS2040S context. Perhaps consider parsing /mod and /lec arguments to determine context instead. This would allow user to specify their context directly for the find command i.e. `find Week 1 Week2 /mod CS2040S` to search for lectures in CS2040S that match the "Week 1" and "Week2" keywords.

Reply:
Where is the code logic for the AddCommand having such capability?

Reply:
Its okay I've figured it out, will make changes to my code and commit them later